docs/DafnyRef/GrammarChecker.java

```diff
@@ -0,0 +1,47 @@
+public class GrammarChecker {
+
+    public static void main(String ... args) {
+        String grammarFile = args[0];
+        String docGrammarFile = args[1];
+        String inputGrammar = Files.readAllBytes(grammarFile);
+
+        source(inputGrammar).removeComments()
+    }
+
+    public static abstract class CharSupplier {
+        Character next();
+        boolean more();
+
+        public CharSupplier removeComments() {
+            int index = 0;
+            Character lookahead = null;
+            new CharSupplier() {
+                Character next() {
+                    r = lookahead;
+                    lookahead = null;
+                    return r;
+                }
+                boolean more() {
+                    if (lookahead == null) {
+                        lookahad = this.
```

Comment:
This doesn't look right, nor does the enclosing class do what it claims... Is this an old version of the file?